devportal-component-relationships/README.md

```diff
@@ -0,0 +1,87 @@
+
+We currently present three types of releationships already:
+
+- `ownedBy` / `ownerOf` between Component and Group, informing about the ownership if a software component.
+
+- `memberOf` / `hasMember` between a Group and a User entity, showing who belongs to which team(s).
+
+- `parentOf` / `childOf`: A parent/child relation to build up a tree, used in our case to describe the organizational structure between Groups. This could also be used for other entity kinds.
+
+Backstage offers the following additional relationship types, which we don't use yet:
+
+- `dependsOn` / `dependencyOf`: A relation denoting a dependency on another entity. This relation is a general expression of being in need of that other entity for an entity to function. It can for example be used to express that a website component needs a library component as part of its build, or that a service component uses a persistent storage resource.
+
+- `partOf` / `hasParts`: A relation with a Domain, System or Component entity, typically from a Component, API, or System. These relations express that a component belongs to a larger component; a component, API or resource belongs to a system; or that a system is grouped under a domain.
+
+Especially for entities of kind API:
+
+- `consumesApi` / `apiConsumedBy`: A relation with an API entity, typically from a Component. These relations express that a component consumes an API - meaning that it depends on endpoints of the API.
+
+- `providesApi` / `apiProvidedBy` is a relationship to express that a certain API (which is another type of catalog entity) is provided by a certain Component.
+
+## Proposal
+
+## Components depending on libraries
+
+Let's have the dependency relationship between components and the libraries they are using visible in the portal. Here we refer only to the libraries we own (in Go that's any module named `github.com/giantswarm/*`), since adding entities for the thousands of third party libraries we use would bloat our catalog heavily and probably make it unusable.
+
+In a Go project, the fact that a module depends on another module is encoded in the `go.sum` file. Alternatively, we can also investigate the [Github API for exorting a bill of materials](https://docs.github.com/en/rest/dependency-graph/sboms?apiVersion=2022-11-28).
+
+Other languages we use (Python, JavaScript/Typescript, ...) have similar concepts and may be explored, however each of those is only a niche in our landscape and we will have to evaluate whether the value gained is worth the effort.
+
+## Components depending on other components (services)
+
+We have cases where a certain service can only function when another service is also deployed with it. Examples:
+
+- `kubectl-gs` (CLI) depends on `athena` (service) running in whatever cluster to sign in to.
```

Comment:
This is only true for parts of `kubectl-gs` actually, `login` I assume. You could say the same for `template` to depend on `Vault` or `SOPS` key being present, but those also depend on installation.

Reply:
Yes, this example is for the `login` command. Adding clarification.